[Identity] Allowing to specify a user defined Managed Identity by Resource Id (#20043)
Azure/azure-sdk-for-js: sdk/identity/identity/CHANGELOG.md

### Features Added

> - Added support to specify a custom resource Id on the `ManagedIdentityCredential` and the `DefaultAzureCredential`.

Reviewer: ```suggestion
- Added support for specifying a custom `resourceId` when creating a `ManagedIdentityCredential` or `DefaultAzureCredential`.
```

> - Added support to specify a custom resource Id on the `ManagedIdentityCredential` and the `DefaultAzureCredential`.

Author: Thank you!

  - In scenarios such as when user assigned identities are created using an ARM template, where the resource Id of the identity is known but the client Id can't be known ahead of time, this parameter allows programs to use these user assigned identities without having to first determine the client Id of the created identity.
  - If `resourceId` is sent, the managed identities that don't support this parameter will be skipped.

> - Added `clientId` to the optional parameters of the `ManagedIdentityCredential`.

Author: good point. You mentioned to make the types clearer for this purpose. I took a note, but I haven’t followed up on it.

> - Added `clientId` to the optional parameters of the `ManagedIdentityCredential`.

Author: How does this look? https://github.com/Azure/azure-sdk-for-js/pull/20043/commits/a4857a64b21b7d2b430cb81513e7daffb57c57e3

> - Added `clientId` to the optional parameters of the `ManagedIdentityCredential`.

Reviewer: This looks good, thanks!
